### DISCUSSION TITLE
Jacobi solver with HIP and OpenMP: Fortran examples with HIP and OpenMP

### DISCUSSION THREAD

**Discussion by AlexisEspinosaGayosso** [amd/amd-lab-notes#15] [General]
In your Jacobi solver with HIP and OpenMP, you mention at the end that you are working on the Fortran-OpenMP example as additional material for the lab. I'm really looking forward for it but, could you also add a HIP-Fortran example as part of the material too? I think this is very important!

**Comment by jychang48** [comment]
@AlexisEspinosaGayosso what specifically about HIP-Fortran are you looking for? If you're looking for ways to interface Fortran to HIP, we recommend using hipfort: https://github.com/ROCmSoftwarePlatform/hipfort

They also have some simple examples, including both the HIP/ROCm math libraries or plain HIP kernels themselves: https://github.com/ROCmSoftwarePlatform/hipfort/tree/develop/test

We have had to write our own Fortran to C to HIP interfaces for some applications that relied on Cray compilers, but either way blog material relating to Fortran may not come until later next year.

**Comment by AlexisEspinosaGayosso** [comment]
Thanks a lot Justin,

I will carefully read the links you are providing. I'm sure they will be very useful.

In regards to my initial suggestion, I was just looking for a "completion" of your example. You already "advertised" a future addition to this exercise with the use of Fortran+OpenMP. Then, I thought that the complete topic should have also the HIP-Fortran piece. And I think it's important to consider my suggestion plan it for the near near future.

This addition, together with the links you have provided above, will give a more complete information to users, I think.

Thanks a lot for your interest and support.

**Reply by jychang48** [reply]
Sure, we can have a HIP-Fortran piece to complement the eventual Fortran OpenMP work. I'll let the authors working on this know that there's a request for this
